test/TensorFlowRuntime/description.swift

```diff
@@ -0,0 +1,250 @@
+        [1, 1, 1, ..., 1, 1, 1],
+        ...,
+        [1, 1, 1, ..., 1, 1, 1],
+        [1, 1, 1, ..., 1, 1, 1],
+        [1, 1, 1, ..., 1, 1, 1]],
+
+       [[1, 1, 1, ..., 1, 1, 1],
+        [1, 1, 1, ..., 1, 1, 1],
+        [1, 1, 1, ..., 1, 1, 1],
+        ...,
+        [1, 1, 1, ..., 1, 1, 1],
+        [1, 1, 1, ..., 1, 1, 1],
+        [1, 1, 1, ..., 1, 1, 1]],
+
+       [[1, 1, 1, ..., 1, 1, 1],
+        [1, 1, 1, ..., 1, 1, 1],
+        [1, 1, 1, ..., 1, 1, 1],
+        ...,
+        [1, 1, 1, ..., 1, 1, 1],
+        [1, 1, 1, ..., 1, 1, 1],
+        [1, 1, 1, ..., 1, 1, 1]],
+
+       ...,
+
+       [[1, 1, 1, ..., 1, 1, 1],
+        [1, 1, 1, ..., 1, 1, 1],
+        [1, 1, 1, ..., 1, 1, 1],
+        ...,
+        [1, 1, 1, ..., 1, 1, 1],
+        [1, 1, 1, ..., 1, 1, 1],
+        [1, 1, 1, ..., 1, 1, 1]],
+
+       [[1, 1, 1, ..., 1, 1, 1],
+        [1, 1, 1, ..., 1, 1, 1],
+        [1, 1, 1, ..., 1, 1, 1],
+        ...,
+        [1, 1, 1, ..., 1, 1, 1],
+        [1, 1, 1, ..., 1, 1, 1],
+        [1, 1, 1, ..., 1, 1, 1]],
+
+       [[1, 1, 1, ..., 1, 1, 1],
+        [1, 1, 1, ..., 1, 1, 1],
+        [1, 1, 1, ..., 1, 1, 1],
+        ...,
+        [1, 1, 1, ..., 1, 1, 1],
+        [1, 1, 1, ..., 1, 1, 1],
+        [1, 1, 1, ..., 1, 1, 1]]]
+      """, tensor.description)
+  }
+}
+
+// Test example random initialization (longer scalars).
+// Hard-coded test tensors generated via `Tensor(randomNormal:)`.
+DescriptionTests.test("RandomScalars") {
+  do {
+    var matrix = Tensor<Float>(
+      [[  1.1621541, -0.39326498],
+       [ -1.5391855,    -1.11794]]
+    )
+    expectEqual("""
+      [[  1.1621541, -0.39326498],
+       [ -1.5391855,    -1.11794]]
+      """, matrix.description)
+    // Use one short scalar.
+    // Check that scalars are still aligned (properly left padded).
+    matrix[0][1] = Tensor(-1)
+    expectEqual("""
+      [[ 1.1621541,       -1.0],
+       [-1.5391855,   -1.11794]]
+      """, matrix.description)
+  }
+
+  do {
+    var matrix = Tensor<Double>(
+      [[  0.3374860907024739,   0.1805635511143933],
+       [  0.1861132028235124, -0.08243178459215775]]
+    )
+    expectEqual("""
+      [[  0.3374860907024739,   0.1805635511143933],
+       [  0.1861132028235124, -0.08243178459215775]]
+      """, matrix.description)
+    // Use one short scalar.
+    // Check that scalars are still aligned (properly left padded).
+    matrix[0][1] = Tensor(-1)
+    expectEqual("""
+      [[  0.3374860907024739,                 -1.0],
+       [  0.1861132028235124, -0.08243178459215775]]
+      """, matrix.description)
+  }
+}
+
+DescriptionTests.test("DescriptionConfiguration") {
+  // Test `lineWidth` configuration.
+  let vector = Tensor<Float>(ones: [10])
+  expectEqual("[1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0]",
+              vector.description)
+  expectEqual("""
+    [1.0, 1.0,
+     1.0, 1.0,
+     1.0, 1.0,
+     1.0, 1.0,
+     1.0, 1.0]
+    """, vector.description(lineWidth: 6))
```

Comment:
Line width configuration doesn't force the description to have a max line width, which may be unexpected.
The line width logic is ported directly from PyTorch (which is heavily NumPy-inspired), so I wonder if line width configuration exhibits the same behavior.

The relevant calculation is:
```swift
    let maxScalarLength =
      scalars.lazy.map { String(describing: $0).count }.max() ?? 3
    let scalarCountPerLine = Swift.max(1, lineWidth / maxScalarLength)
```
In this test, `lineWidth = 6` and `maxScalarLength = 3`, so `scalarCountPerLine = 2`.
Two scalars are printed per line, but the line width exceeds 6 characters.